hikari/impl/bot.py

```diff
@@ -1049,3 +1052,20 @@ def _validate_activity(activity: undefined.UndefinedNoneOr[presences.Activity])
                 category=errors.HikariWarning,
                 stacklevel=3,
             )
+
+
+class BotApp(GatewayBot):
+    """Deprecated alias for `GatewayBot`.
+
+    .. deprecated:: 2.0.0.dev101
+        Use `GatewayBot` instead.
+    """
+
+    # This is only overridden at runtime in-order to preserve this class'
+    # inherited init pattern from `GatewayBot` without repeating it
+    if not typing.TYPE_CHECKING:
+
+        def __init__(self, *args: typing.Any, **kwargs: typing.Any) -> None:
+            super().__init__(*args, **kwargs)
+            # This is called after the super call as __init__ may alter the warnings config
+            deprecation.warn_deprecated(BotApp, stack_level=3)
```

Comment:
```suggestion
@deprecation.deprecated(version="2.0.0.dev102", alternative="GatewayBot")
class BotApp(GatewayBot):
    """Alias for `GatewayBot`."""
```

Reply:
That would both be a breaking change and also break the typing for anybody who's accessing stuff on the BotApp class like unbound methods